pkg/sdk/secrets_gen_test.go

```diff
@@ -0,0 +1,321 @@
+	t.Run("validation: invalid identifier", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.name = emptySchemaObjectIdentifier
+		assertOptsInvalidJoinedErrors(t, opts, ErrInvalidObjectIdentifier)
+	})
+
+	t.Run("validation: conflicting fields for [opts.OrReplace opts.IfNotExists]", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.IfNotExists = Bool(true)
+		opts.OrReplace = Bool(true)
+		assertOptsInvalidJoinedErrors(t, opts, errOneOf("CreateWithGenericStringSecretOptions", "OrReplace", "IfNotExists"))
+	})
+
+	t.Run("all options", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.IfNotExists = Bool(true)
+		opts.SecretString = "test"
+		assertOptsValidAndSQLEquals(t, opts, "CREATE SECRET IF NOT EXISTS %s TYPE = GENERIC_STRING SECRET_STRING = 'test'", id.FullyQualifiedName())
+	})
+}
+
+func TestSecrets_Alter(t *testing.T) {
+	id := randomSchemaObjectIdentifier()
+
+	defaultOpts := func() *AlterSecretOptions {
+		return &AlterSecretOptions{
+			name: id,
+		}
+	}
+
+	setOpts := func() *AlterSecretOptions {
+		return &AlterSecretOptions{
+			name:     id,
+			Set:      &SecretSet{},
+			IfExists: Bool(true),
+		}
+	}
+
+	unsetOpts := func() *AlterSecretOptions {
+		return &AlterSecretOptions{
+			name:     id,
+			Unset:    &SecretUnset{},
+			IfExists: Bool(true),
+		}
+	}
+
+	t.Run("validation: nil options", func(t *testing.T) {
+		var opts *AlterSecretOptions = nil
+		assertOptsInvalidJoinedErrors(t, opts, ErrNilOptions)
+	})
+
+	t.Run("validation: exactly one field from [opts.Set opts.Unset] should be present", func(t *testing.T) {
+		opts := defaultOpts()
+		assertOptsInvalidJoinedErrors(t, opts, errExactlyOneOf("AlterSecretOptions", "Set", "Unset"))
+	})
+
+	t.Run("validation: exactly one field from [opts.Set.SetForOAuthClientCredentialsFlow opts.Set.SetForOAuthAuthorizationFlow opts.Set.SetForBasicAuthentication opts.Set.SetForGenericString] should be present", func(t *testing.T) {
+		opts := setOpts()
+		opts.Set.SetForOAuthAuthorizationFlow = &SetForOAuthAuthorizationFlow{}
+		opts.Set.SetForBasicAuthentication = &SetForBasicAuthentication{}
+		assertOptsInvalidJoinedErrors(t, opts, errExactlyOneOf("AlterSecretOptions.Set", "SetForOAuthClientCredentialsFlow", "SetForOAuthAuthorizationFlow", "SetForBasicAuthentication", "SetForGenericString"))
+	})
+
+	t.Run("alter: set options for Oauth Client Credentials Flow", func(t *testing.T) {
+		opts := setOpts()
+		opts.Set.Comment = String("test")
+		opts.Set.SetForOAuthClientCredentialsFlow = &SetForOAuthClientCredentialsFlow{[]SecurityIntegrationScope{{"sample_scope"}}}
+		assertOptsValidAndSQLEquals(t, opts, "ALTER SECRET IF EXISTS %s SET COMMENT = 'test' OAUTH_SCOPES = ('sample_scope')", id.FullyQualifiedName())
+	})
+
+	t.Run("alter: set options for Oauth Authorization Flow", func(t *testing.T) {
+		opts := setOpts()
+		opts.Set.Comment = String("test")
+		opts.Set.SetForOAuthAuthorizationFlow = &SetForOAuthAuthorizationFlow{
+			String("test_token"),
+			String("2024-11-11"),
```

Comment:
Hmm, I don't know why linter was skipped here, but it always shouts not to use unnamed fields like that 🤔

Reply:
No clue either, but it shouted afterwards so the fields had been named